### PR TITLE
Caller: use lonely operator to safely call `#first`

### PR DESCRIPTION
**Pull request by chastell**
The Caller cop currently replaces `caller[42]` with `caller(43..43).first`.

If the caller stack is short, the former returns `nil` while the latter raises a `NoMethodError` (_undefined method `first' for nil_).

This PR fixes this by using the lonely operator (`&.`) to call `#first` only if the result is non-`nil`.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop-performance/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by koic** [comment]
Thank you for the suggestion. I understand the reasoning, but this solution cannot be adopted for two reasons:

- Safe navigation is not always necessary. In cases where it is certainly unnecessary, it becomes redundant.  
- In situations where an error is preferable to incorrect behavior, this approach would result in unintended behavior. Unexpected behavior is more problematic than an explicit error.  

In other words, while safe navigation is appropriate in the context of this PR, there are cases where it is not.
